src/instana/instrumentation/couchbase_inst.py

```diff
@@ -6,90 +6,148 @@
-                  'observe_multi', 'endure_multi', 'remove_multi', 'counter_multi']
-
-    def capture_kvs(scope, instance, query_arg, op):
+    operations = [
+        "upsert",
+        "insert",
+        "replace",
+        "append",
+        "prepend",
+        "get",
+        "rget",
+        "touch",
+        "lock",
+        "unlock",
+        "remove",
+        "counter",
+        "mutate_in",
+        "lookup_in",
+        "stats",
+        "ping",
+        "diagnostics",
+        "observe",
+        "upsert_multi",
+        "insert_multi",
+        "replace_multi",
+        "append_multi",
+        "prepend_multi",
+        "get_multi",
+        "touch_multi",
+        "lock_multi",
+        "unlock_multi",
+        "observe_multi",
+        "endure_multi",
+        "remove_multi",
+        "counter_multi",
+    ]
+
+    def collect_attributes(
+        span: InstanaSpan,
+        instance: Bucket,
+        query_arg: Any,
+        op: str,
+    ) -> None:
         try:
-            scope.span.set_tag('couchbase.hostname', instance.server_nodes[0])
-            scope.span.set_tag('couchbase.bucket', instance.bucket)
-            scope.span.set_tag('couchbase.type', op)
+            span.set_attribute("couchbase.hostname", instance.server_nodes[0])
+            span.set_attribute("couchbase.bucket", instance.bucket)
+            span.set_attribute("couchbase.type", op)
 
-            if query_arg is not None:
+            if query_arg:
                 query = None
                 if type(query_arg) is N1QLQuery:
                     query = query_arg.statement
                 else:
                     query = query_arg
 
-                scope.span.set_tag('couchbase.sql', query)
-        except:
+                span.set_attribute("couchbase.sql", query)
+        except Exception:
             # No fail on key capture - best effort
             pass
 
-    def make_wrapper(op):
-        def wrapper(wrapped, instance, args, kwargs):
+    def make_wrapper(op: str) -> Callable:
+        def wrapper(
+            wrapped: Callable[..., object],
+            instance: couchbase.bucket.Bucket,
+            args: Tuple[object, ...],
+            kwargs: Dict[str, Any],
+        ) -> Any:
             tracer, parent_span, _ = get_tracer_tuple()
+            parent_context = parent_span.get_span_context() if parent_span else None
 
             # If we're not tracing, just return
             if tracing_is_off():
                 return wrapped(*args, **kwargs)
 
-            with tracer.start_active_span("couchbase", child_of=parent_span) as scope:
-                capture_kvs(scope, instance, None, op)
+            with tracer.start_as_current_span(
+                "couchbase", span_context=parent_context
+            ) as span:
+                collect_attributes(span, instance, None, op)
                 try:
                     return wrapped(*args, **kwargs)
                 except Exception as e:
```

Comment:
Use the following expression along your file (and all other instrumentation) and skip this block for coverage counting:
```suggestion
                except Exception as exc:
```